src/pyclass.rs

```diff
@@ -51,6 +51,8 @@ pub(crate) unsafe fn default_new<T: PyTypeInfo>(
         }
         #[cfg(Py_LIMITED_API)]
         {
+            // Silence unused parameter warning.
+            let _ = py;
```

Comment:
Could this be solved in a cleaner manner by splitting the function into `#[cfg(PY_LIMITED_API)]` and `#[cfg(not(PY_LIMITED_API))]` fns? This method is currently rather hard to read!